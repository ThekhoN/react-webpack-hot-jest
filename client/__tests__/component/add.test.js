import React from 'react';
import Add from '../../src/component/add';
import renderer from 'react-test-renderer';
import {shallow, mount} from 'enzyme';

test('Add Component matches snapshot', () => {
  const component = renderer.create(<Add/>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

// test behaviour by mounting component
describe('Add', function () {
  let onAdd;
  let add;

  beforeEach(() => {
    // jest.fn() creates a mock function
    onAdd = jest.fn();
    add = mount(<Add onAdd={onAdd} />);
  });
  it('Add requires onAdd prop', () => {
    expect(add.props().onAdd).toBeDefined();
  });
  it('Add renders button', () => {
    const button = add.find('button').first();
    expect(button).toBeDefined();
  });
  it('Button click calls onAdd', () => {
    const button = add.find('button').first();
    const input = add.find('input').first();
    input.simulate('change', {target: {value: 'Name 4'}});
    button.simulate('click');
    expect(onAdd).toBeCalledWith('Name 4');
  });
});
