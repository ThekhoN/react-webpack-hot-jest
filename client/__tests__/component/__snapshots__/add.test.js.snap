// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Component matches snapshot 1`] = `
<form>
  <input
    name="name"
    onChange={[Function]}
    type="text"
    value=""
  />
  <button
    onClick={[Function]}
  >
    Add
  </button>
</form>
`;
